File: src/tasks/model/TaskBuilder.java
package tasks.model;

public interface TaskBuilder {

  TaskBuilder setStatus(Task.Status status);

  TaskBuilder setBlockingTasks(Iterable<? extends Task> task);

  TaskBuilder addBlockingTask(Task task);

  TaskBuilder setBlockedTasks(Iterable<? extends Task> task);

  TaskBuilder addBlockedTask(Task task);
}

File: src/tasks/model/TaskBuilder.kt
package tasks.model

interface TaskBuilder {
  fun setStatus(status: Task.Status): TaskBuilder
  fun setBlockingTasks(tasks: Iterable<Task>): TaskBuilder
  fun addBlockingTask(task: Task): TaskBuilder
  fun setBlockedTasks(tasks: Iterable<Task>): TaskBuilder
  fun addBlockedTask(task: Task): TaskBuilder
}